app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawer.kt

```diff
@@ -0,0 +1,232 @@
+            var endTimePreviousLecture: Int = conference.firstEventStartsAt
+            var startTime: Int
+            var margin: Int
+            var previousLecture: Lecture? = null
+            val lectureLayoutParams = mutableMapOf<Lecture, LinearLayout.LayoutParams>()
+
+            for (idx in lectures.indices) {
+                val lecture = lectures[idx]
+
+                if (lecture.roomIndex == roomIndex) {
+                    if (lecture.dateUTC > 0) {
+                        startTime = Moment(lecture.dateUTC).minuteOfDay
+                        if (startTime < endTimePreviousLecture) {
+                            startTime += Duration.ofDays(1).toMinutes().toInt()
+                        }
+                    } else {
+                        startTime = lecture.relStartTime
+                    }
+                    if (startTime > endTimePreviousLecture) {
+                        margin = standardHeight * (startTime - endTimePreviousLecture) / DIVISOR
+                        val isConsecutive = previousLecture != null
+                        if (isConsecutive) {
+                            lectureLayoutParams[previousLecture]!!.bottomMargin = margin
+                            margin = 0
+                        }
+                    } else {
+                        // first lecture
+                        margin = 0
+                    }
+
+                    // fix overlapping events
+                    var next: Lecture? = null
+                    for (nextIndex in idx + 1 until lectures.size) {
+                        next = lectures[nextIndex]
+                        if (next.roomIndex == roomIndex) {
+                            break
+                        }
+                        next = null
+                    }
+                    if (next != null) {
+                        if (next.dateUTC > 0) {
+                            val endTimestamp = lecture.dateUTC + lecture.duration * MILLIS_PER_MINUTE
+                            // next starts, before current ends
+                            if (endTimestamp > next.dateUTC) {
+                                Log.d(LOG_TAG, "${lecture.title} collides with ${next.title}")
+                                // cut current at the end, to match next lectures start time
+                                lecture.duration = ((next.dateUTC - lecture.dateUTC) / MILLIS_PER_MINUTE).toInt()
+                            }
+                        }
+                    }
+
+                    if (!lectureLayoutParams.containsKey(lecture)) {
+                        lectureLayoutParams[lecture] = LinearLayout.LayoutParams(0, 0)
+                    }
+                    lectureLayoutParams[lecture]!!.topMargin = margin
+                    endTimePreviousLecture = startTime + lecture.duration
+                    previousLecture = lecture
+                }
+            }
+
+            return lectureLayoutParams
```

Comment:
Instead of returning the mutable map you could return an immutable map via `return lectureLayoutParams.toMap()`.

Reply:
MutableMap.toMap still returns a mutable map as far as I can see. Hence I leave it as is.

Reply:
The documentation states:

> Returns a new read-only map containing all key-value pairs from the original map.

I consider this a good way to protect the original map from modifications.

Reply:
Yes I read that. But the code tells me another story:
<img width="683" alt="EventFahrplan____repos_EventFahrplan__-____gradle_caches_modules-2_files-2_1_org_jetbrains_kotlin_kotlin-stdlib-common_1_3_61_fe3b9d48f4c3947639b6b1760213c1b79281383e_kotlin-stdlib-common-1_3_61-sources_jar__kotlin_collections_Maps_kt__Grad" src="https://user-images.githubusercontent.com/1071386/79053626-17111200-7c3f-11ea-8dd4-6f3fa165dfef.png">

Reply:
:exploding_head: ... looks like you found a bug in the documentation. This opens up an opportunity for you [contribute to Kotlin](https://androidessence.com/contributing-to-kotlin) \o/

With regards to the case I raised I now think that it is safe to return a mutable list to the caller since it does not matter to `LectureViewDrawer#calculateLayoutParams` "anymore" if the list is modified at a later point since the function is "done with it". - I therefore close this issue here.